06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,216 @@
+/**
+ *
+ * @class InvertedIndex
+ */
+class InvertedIndex {
+
+  /**
+   * Creates an instance of InvertedIndex.
+   * @memberOf InvertedIndex
+   */
+  constructor() {
+    this.searchIndices = {};
+    this.indexedFiles = {};
+  }
+
+  /**
+   *
+   *
+   * @param {Object} file - Javascript object containing file properties
+   * @returns {Object} response - JSON content of the read file
+   *
+   * @memberOf InvertedIndex
+   */
+  static readFile(file) {
+    let fileContent;
+    return new Promise((resolve, reject) => {
+      try {
+        const reader = new FileReader();
+        reader.onload = (event) => {
+          fileContent = JSON.parse(event.target.result);
+          const response = InvertedIndex.validateFile(fileContent);
+          resolve(response);
+        };
+        reader.readAsText(file);
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+
+  /**
+   *
+   *
+   * @param {Object} jsonContent - file content
+   * @returns{Object} isValid - returns true and JSON content for valid files
+   * and false for invalid files
+   * @memberOf InvertedIndex
+   */
+  static validateFile(jsonContent) {
+    let isValid = { status: false };
+    const invalidStructureError = {
+      name: 'validate file structure',
+      message: 'File structure is invalid'
+    };
+    if (Object.keys(jsonContent).length === 0 &&
+       typeof jsonContent === 'object') {
+      return isValid;
+    }
+    try {
+      jsonContent.forEach((doc) => {
+        if (!doc.title || !doc.text) {
+          throw invalidStructureError;
+        } else {
+          isValid = {
+            status: true,
+            jsonContent
+          };
+          return isValid;
+        }
+      });
+    } catch (error) {
+      if (error.name === 'validate file structure') return isValid;
+      throw error;
+    }
+    return isValid;
+  }
```

Comment:
I tried your [application](https://book-switch-staging.herokuapp.com/) and It appears you are not validating that the content is properly formatted. Please fix that and also provide tests for that too.